### PR TITLE
fix(std): nixago editorconfig special ini format

### PR DESCRIPTION
**Pull request by blaggacao**
fix: #188

### CODE REVIEW

File: cells/std/nixago/editorconfig.nix [hunk 1]
```diff
@@ -2,10 +2,19 @@
...
   inherit (inputs) nixpkgs;
 in {
   configData = {};
   output = ".editorconfig";
-  format = "ini";
+  engine = request: let
+    inherit (request) configData output;
+    name = l.baseNameOf output;
+    value = {
+      globalSection = {root = configData.root or false;};
```
Comment (blaggacao) [inline cells/std/nixago/editorconfig.nix:14]: If not set, defaults explicitly to `false` which is cool, I guess.

Comment (montchr) [reply cells/std/nixago/editorconfig.nix:14]: I think `true` would be a better default here.

In my experience across many repos, a proverbial 98% of them only have a single `.editorconfig` file at the project root, where `root = true` would apply. Even in monorepos including a few different languages and sub-projects with their own distinct scope, each of these scopes will likely adhere to the same coding standards declared in the root-level `.editorconfig`. So, in those proverbial 2% of projects where we've had multiple of these files, the additional files turned out to be an unnecessary source of confusion in the end.

File: cells/std/nixago/editorconfig.nix [hunk 2]
```diff
@@ -2,10 +2,19 @@
...
+      sections = l.removeAttrs configData ["root"];
```
Comment (montchr) [inline cells/std/nixago/editorconfig.nix:15]: Interesting -- this reminds me of an oddity I've noticed in a `treefmt.toml` generated with Standard:

```toml
[formatters] # <- unnecessary, but harmless (i think)
[formatters.nix]
command = "alejandra"
...
```

That said, the additional section could have been something I introduced in my own project accidentally...

So I take it this removal of the `root` section prevents something similar?

**Review by montchr — APPROVED** [review]
LGTM, with one suggestion

File: cells/std/nixago/editorconfig.nix (hunk 1 above)
Comment (montchr) [reply cells/std/nixago/editorconfig.nix:14]: of course, all that said, i don't recall ever having noticed a difference with having `root = true` or `root = false` or omitting it entirely 🤷‍♀️

File: cells/std/nixago/editorconfig.nix (hunk 2 above)
Comment (blaggacao) [reply cells/std/nixago/editorconfig.nix:15]: Yeah, this is just the way that nix' `toTOML` is implemented.

**Pull request merged**